tensorboard/webapp/widgets/line_chart_v2/sub_view/line_chart_interactive_view_test.ts

```diff
@@ -0,0 +1,368 @@
+    fixture: ComponentFixture<TestableComponent>,
+    eventName: 'wheel',
+    eventInit: WheelEventInit
+  ): void;
+  function emitEvent(
+    fixture: ComponentFixture<TestableComponent>,
+    eventName:
+      | 'mousedown'
+      | 'mouseup'
+      | 'mousemove'
+      | 'mouseenter'
+      | 'mouseleave'
+      | 'dblclick',
+    eventInit: MouseEventInit
+  ): void;
+  function emitEvent(
+    fixture: ComponentFixture<TestableComponent>,
+    eventName: string,
+    eventInit: MouseEventInit | WheelEvent
+  ) {
+    const dom = fixture.debugElement.query(By.css('svg'))
+      .nativeElement as SVGElement;
+    const event =
+      eventName === 'wheel'
+        ? new WheelEvent(eventName, {relatedTarget: dom, ...eventInit})
+        : new MouseEvent(eventName, {relatedTarget: dom, ...eventInit});
+    dom.dispatchEvent(event);
+  }
+
+  beforeEach(async () => {
+    await TestBed.configureTestingModule({
+      declarations: [LineChartInteractiveViewComponent, TestableComponent],
+      imports: [CommonModule, OverlayModule],
+    }).compileComponents();
+
+    overlayContainer = TestBed.inject(OverlayContainer);
+
+    onViewExtentChange = jasmine.createSpy();
+    onViewExtentReset = jasmine.createSpy();
+  });
+
+  describe('tooltips', () => {
+    it('shows tooltip when moving mouse inside the container', () => {
+      const fixture = createComponent();
+      fixture.detectChanges();
+
+      expect(overlayContainer.getContainerElement().childElementCount).toBe(0);
+
+      emitEvent(fixture, 'mouseenter', {clientX: 10, clientY: 10});
+      fixture.detectChanges();
+
+      expect(overlayContainer.getContainerElement().childElementCount).toBe(1);
+    });
+
+    it('renders display name and information about the series close to cursor', () => {
+      const fixture = createComponent();
+      fixture.componentInstance.seriesData = [
+        createSeries('foo', (index: number) => index),
+        createSeries('bar', (index: number) => index * 2),
+      ];
+      fixture.componentInstance.seriesMetadataMap = {
+        foo: buildMetadata({id: 'foo', displayName: 'Foo', color: '#f00'}),
+        bar: buildMetadata({id: 'bar', displayName: 'Bar name', color: '#00f'}),
+      };
+      fixture.componentInstance.domDim = {width: 500, height: 200};
+      fixture.detectChanges();
+
+      emitEvent(fixture, 'mouseenter', {clientX: 250, clientY: 10});
+      fixture.detectChanges();
+
+      const [
+        foo,
+        bar,
+      ] = overlayContainer.getContainerElement().querySelectorAll('tbody tr');
+
+      expect(
+        (foo.querySelector('.circle span') as HTMLSpanElement).style
+          .backgroundColor
+      ).toBe('rgb(255, 0, 0)');
+      expect(
+        (bar.querySelector('.circle span') as HTMLSpanElement).style
+          .backgroundColor
+      ).toBe('rgb(0, 0, 255)');
+
+      expect(
+        [...foo.querySelectorAll('td:not(.circle)')].map((td) => td.textContent)
+      ).toEqual(['Foo', '5', '5']);
```

Comment:
Ah, I understood why these are '5' after reading, but perhaps we can add a short comment somewhere like:
"The mouse is in the middle of the series with values 0 - 9."